tests/components/prometheus/helpers.py

```diff
@@ -0,0 +1,170 @@
+"""Helper methods for prometheus tests."""
+
+
+class MetricsTestHelper:
+    """Helps with formatting prometheus metrics and future-proof label changes."""
+
+    @classmethod
+    def get_device_class_label_line(cls, device_class):
+        """Format the device_class label line (tries to account for enum class)."""
+        if device_class:
+            device_class = str(device_class)
+        return f'device_class="{device_class or ""}",'
+
+    @classmethod
+    def _get_metric_string(
+        cls,
+        metric_name,
+        domain,
+        friendly_name,
+        object_id,
+        metric_value=None,
+        area=None,
+        device_class=None,
+    ):
+        device_class_label_line = cls.get_device_class_label_line(device_class)
+        final_metric_value = f" {metric_value}" if metric_value else ""
+        full_metric_string = (
+            f"{metric_name}{{"
+            f'area="{area or ""}",'
+            f"{device_class_label_line}"
+            f'domain="{domain}",'
+            f'entity="{domain}.{object_id}",'
+            f'friendly_name="{friendly_name}",'
+            f'object_id="{object_id}"'
+            f"}}{final_metric_value}"
+        )
+        return full_metric_string
+
+    @classmethod
+    def _perform_metric_assert(
+        cls,
+        metric_name,
+        metric_value,
+        domain,
+        friendly_name,
+        object_id,
+        body,
+        area=None,
+        device_class=None,
+        positive_comparison=True,
+    ):
+        full_metric_string = cls._get_metric_string(
+            metric_name,
+            domain,
+            friendly_name,
+            object_id,
+            area=area,
+            device_class=device_class,
+            metric_value=metric_value,
+        )
+        if positive_comparison:
```

Comment:
It might be more convenient to provide separate assertion functions (e.g. `assert_metric_present` vs `assert_metric_absent`) rather than make the user set a bool.